Dependencies warning fix: remote discovery for mod update and mod prune (#2178)
bufbuild/buf: CHANGELOG.md

## [Unreleased]

> - Remove an import warning in case the import statement points to a local unnamed module.

Reviewer: Remove

> - Remove an import warning in case such import statement points to a local unnamed module.

Reviewer 2: ```suggestion
- Remove an import warning in case the import points to a local unnamed module.
```

> - Discover the correct remote to use for fetching dependencies in the `mod update` and `mod prune`

Reviewer: Remove

>   commands, in case it's an unnamed module and there is a mix of remotes in the `buf.yaml`

Reviewer 2: ```suggestion
  commands, in case it's an unnamed module and there are a mix of remotes in the `buf.yaml`
```
